Azure/sonic-buildimage
# [bgpcfgd] Support for FRR configuration based on config DB events

**- Why I did it**
1. Support for non-template based FRR configurations (BGP, route-map, OSPF, static route..etc) using config DB schema.
1. Support for save & restore - Jinja template based config-DB data read and apply to FRR during startup

**- How I did it**

Added a new script and corresponding service. Based on the settings in DEVICE_METADATA table, new service will be started along with the bgp container start. And when user changed the BGP or other related table entries in config DB, the daemon will run corresponding VTYSH commands to program on FRR.
Also added jinja template to generate FRR config file to be used by FRR daemons while bgp container restarted

**- How to verify it**
1. Add/delete data on config DB and then run VTYSH "show running-config" command to check if FRR configuration changed.
1. Restart bgp container and check if generated FRR config file is correct and run VTYSH "show running-config" command to check if FRR configuration is consistent with attributes in config DB

**- Description for the changelog**



**- A picture of a cute animal (not mandatory but encouraged)**
